packages/sanity/test/descriptors/schema.test.tsx

```diff
@@ -2,25 +2,52 @@ import {
   defineField,
   type FieldGroupDefinition,
   type FieldsetDefinition,
+  type Schema,
   type SchemaTypeDefinition,
 } from '@sanity/types'
+import {cloneDeep} from 'lodash'
 import {type ReactNode} from 'react'
 import {assert, describe, expect, test} from 'vitest'
 
 import {
   type EncodedNamedType,
   type ObjectField,
 } from '../../../@sanity/schema/src/descriptors/types'
+import {extractManifestSchemaTypes} from '../../src/_internal/manifest/extractWorkspaceManifest'
 import {builtinSchema, createSchema, DESCRIPTOR_CONVERTER} from '../../src/core/schema'
+import {normalizeConvertedType, normalizeSchemaType} from './utils'
 
-const findTypeInDesc = (
-  name: string,
-  descriptor: ReturnType<(typeof DESCRIPTOR_CONVERTER)['get']>,
-) =>
+type Descriptor = ReturnType<(typeof DESCRIPTOR_CONVERTER)['get']>
+
+const findTypeInDesc = (name: string, descriptor: Descriptor) =>
   Object.values(descriptor.objectValues).find((val) => val.name === name) as
     | EncodedNamedType
     | undefined
 
+function expectManifestSchemaConversion(schema: Schema, schemaDescriptor: Descriptor) {
+  // Mimic extracting the schema via "sanity deploy"
+  const schemaTypes = extractManifestSchemaTypes(schema) // sanity deploy
+  const data = JSON.parse(JSON.stringify(schemaTypes)) // wire
+  const converted = createSchema({name: schema.name, types: data}) // parsing
```

Comment:
Should we rather have a `normalizeConvertedType` (or similar function) which is called on `data`? We already know that we need some pre-processing due to validations, and I was wondering if the stuff in `normalizeConvertedType` (e.g. handling of "Block" title) should be done in the same phase?